proparse/src/main/java/org/prorefactor/treeparser/TreeParserSymbolScope.java

```diff
@@ -340,14 +339,15 @@ public TableBuffer lookupBuffer(String inName) {
       bufferPart = parts[1];
     }
     TableBuffer symbol = bufferMap.get(bufferPart.toLowerCase());
-    if (symbol == null
-        || (dbPart.length() != 0 && !dbPart.equalsIgnoreCase(symbol.getTable().getDatabase().getName()))) {
+    if (symbol == null || (!dbPart.isEmpty() && !dbPart.equalsIgnoreCase(symbol.getTable().getDatabase().getName()))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1067)